windows-containers/windows-monitoring.md

```diff
@@ -0,0 +1,165 @@
+Windows nodes
+
+### Goals
+
+As part of this enhacement, we plan to do the following:
+* Deploy Windows Management Instrumentation(WMI exporter) onto Windows nodes
+* Configure Prometheus provisioned as part of OpenShift install to 
+  collect data from Windows Nodes
+* Upgrade WMI exporter on the Windows Nodes
+* Leverage cluster-monitoring operator for setting up Prometheus, alert manager
+  and other components
+
+### Non-Goals
+
+As part of this enhacement, we do not plan to do the following:
+* Create a container image for WMI and let WMI run as a container
+* Integrating WMI with cluster monitoring operator
+* Ship Grafana dashboards for Windows Nodes
+
+## Proposal
+
+The main idea here is to run WMI exporter as a Windows Service and let 
+Prometheus instance which was provisioned as part of OpenShift install to 
+collect data from WMI exporter. Since WMI exporter cannot run as a container
+on the Windows node, there by making it impossible to run it within OpenShift cluster, 
+we need to ensure that Prometheus talks to WMI exporter as an external service.
+
+
+### Implementation Details/Notes/Constraints [optional]
+
+We plan to create a Windows Node Monitoring Ansible playbook that can
+* Download WMI exporter Microsoft Installer from a well known location and 
+transfers it to the Windows Node
+* Run WMI msi on the Windows Node
+* Ensure that WMI is running as a Windows Service on the node
+* Create a namespace called `openshift-windows` with 
+  `openshift.io/cluster-monitoring=true` label so that cluster monitoring
+  stack will pick up the service monitor in the next steps.
+* Create service, endpoint in `openshift-windows` pointing to WMI exporter 
+  endpoint
+* Create a service monitor in `openshift-windows` namespace for service
+  created above
+
+The inputs to this playbook
+* The internal ip address of the Windows Node where we want the WMI exporter
+  to run.
+* URL from which WMI msi can be downloaded from
+
+### Justification
+
+The reason for having ansible playbook instead of operator with a container
+image is that all Windows container images contains a Windows Kernel and 
+Red Hat has a policy not to ship 3rd party kernels for support reasons. Please
+look at [alternatives](#Alternatives) section
+
+### Risks and Mitigations
+
+The main risks with this proposal are as follows:
+* We are depending on Microsoft to provide us supported version of WMI. As of 
+  now, WMI is hosted at a repo managed by 
+  [individual](https://github.com/martinlindhe/wmi_exporter).
+  
+  
+Mitigation
+* If Microsoft doesn't provide us with WMI, we would just go ahead with the 
+  upstream repo and start using it.
+
+## Design Details
+
+### Test Plan
+
+We plan to add e2e tests to ensure
+* Service, endpoints and Service Monitors are properly created
+* Prometheus is able to collect data from WMI exporter  
+
+### Graduation Criteria
+
+This enhacement will start as GA
```

Comment:
Totally new features that we have no prior experience with, that immediately go GA are a red flag for me. What speaks against getting at least _some_ experience with these technologies and having it graduate at a later point?

Reply:
At the moment, we've committed to GA as per our discussions with BU but I see your point.